### ISSUE TITLE
The describeIdentity method from the class AWS.CognitoIdentity is not return name-value pairs as the description suggests

### ISSUE THREAD

**Issue by angrychewie** [aws/aws-sdk-js#2073]
Within the Cognito console, the Identity Details page of a specific Identity (under Federated) shows a Date Created, Developer Identifiers, and Linked Logins. When adding a new identity via the sdk, you have to specify a mapping between Developer Identifiers and Linked Logins. In a current project, I need to have a way to grab Developer Identifiers from a given Identity. The description of the describeIdentity method makes it sound like I'd receive a mapping back that's similar in appearance to the one I first sent to create the Identity (which would be perfect). However, instead of a mapping, all I get back is a string array showing only the Linked Logins portion (the half of the map that's useless to me).

If this is intentional, I'd suggest updating the method description. Otherwise, I'd be forever grateful if the method could return an actual mapping between Linked Logins and Developer Identifiers (or even just swap out the Linked Logins string array for a string array of Developer Identifiers!)

Thank you!

**Comment by AllanFly120** [comment]
Hey @angrychewie, sorry for the delayed reply. I'm not very familiar with this API. Can you elaborate what's actually returned from the `describeIdentity()` and what you are expecting?

**Comment by angrychewie** [comment]
A string array of Developer Provider Names is returned. For instance, if angrychewie authenticates via a Federated Identity Pool (not User Pool), that accepted authentication request (with the developer identifier of 'angrychewie') gets 'linked' to the corresponding custom developer provider name that was specified within the request. So in the Cognito console, if you check the identity browser for angrychewie, you'll see 'angrychewie' as the Developer Identifier as well as a field called Linked Login which is just the Developer Provider Name they made the request to (say, login.company.product).

Calling describeIdentity() then just returns a string array of the Developer Provider Names, which in all likelihood could just be 1 thing (login.company.product) such as in my case. 

This is what the documentation states regarding the part of the Data returned:
Logins — (Array<String>)
A set of optional name-value pairs that map provider names to provider tokens

Re-reading this in hindsight, I suppose the method is 'working as intended'.

In the end, though, what really matters to me is having a way to get that Developer Identifier/username back out since my goal is to send it along in the context to other services who will be able to make use of the username, but have no clue about random cognito identities. You need to provide a required logins mapping to make the initial getOpenIdTokenForDeveloperIdentity() call (developer provider name -> developer identifier), so why is there not a way to grab it back out? You can search the Identity Browser in the console using a Developer Identifier, too, and the identifier is clearly shown within a specific identity's details so it's a significant field that's stored somewhere.

I just see no clear way of accessing it and the desribeIdentity() method seems like the most ideal place to have that information be returned (especially when it talks about pairing and mapping). Instead of returning just half of the original map that I initially give it in getOpenIdTokenForDeveloperIdentity() as a string array, I would really like it to return the entire mapping of developer provider name -> developer identifier pairs. 

TLDR: DeveloperIdentifier seems to have been forgotten about even though it's a part of the required mapping parameter in getOpenIdTokenForDeveloperIdentity() and it's a field I could really really use. describeIdentity() already returns half of that original mapping, so why not have it return the useful other half, too?

**Comment by srchase** [comment]
@angrychewie 

I'm opting to close this issue.

This is a service issue and the Cognito Team would be best suited to discuss this in greater depth.

You can post on their [AWS Developer Forum](https://forums.aws.amazon.com/forum.jspa?forumID=173).
